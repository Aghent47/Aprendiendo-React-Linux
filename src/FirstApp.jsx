
// import React, { Fragment } from 'react' = <> 
import PropTypes from 'prop-types';

const getNameUsers = () =>  {
  const users = {
    name: 'Neider Hernández',
    age: '25',
    city: 'Ciudad de México',
   }
  return users;  
}

// Desestructuando los Props {title, subTitle, age}
export const FirstApp = ({ title, subTitle, name}) => { 
 
  return (
    <>
    <h1>{ title }</h1>
    <p> { name } </p>
    {/* <h2> {`${getNameUsers().name} ${age} Años `}</h2>
    <h3> { subtitle } </h3> */}
    <p> { subTitle } </p>
    
    </>
  )
  
}

// Definiendo los PropTypes

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  name: PropTypes.string, 
}

// Definiendo los defaultProps (Se ejecutan primero que los PropTypes)
FirstApp.defaultProps = {
  name: 'Neider Hernández',
  /* title: "Titulo no especificado", */
  subTitle: "Software Engineer",

}