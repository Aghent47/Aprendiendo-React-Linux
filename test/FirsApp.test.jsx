import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas Component <FirstApp/>', () => {

    test('Debe de hacer math con el snapshot', () => {
    const { container } = render(<FirstApp title="Hola, Soy" />)
        
    expect(container).toMatchSnapshot();

    });

    test('debe mostrar el titulo en un <h1></h1>', () => { 
        
        const title = "Hola, soy";
        const { container, getByText } = render(<FirstApp title ={ title }/>)

        expect( getByText(title) ).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
     })
});