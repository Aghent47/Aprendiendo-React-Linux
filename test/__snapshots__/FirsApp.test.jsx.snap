// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Component <FirstApp/> Debe de hacer math con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy
  </h1>
  <p>
     
    Neider Hern√°ndez
     
  </p>
  <p>
     
    Software Engineer
     
  </p>
</div>
`;
